chemprop/cli/common.py

```diff
@@ -148,4 +153,6 @@ def process_common_args(args: Namespace) -> Namespace:
 
 
 def validate_common_args(args):
-    pass
+    # cannot use GPUs if torch cannot detect them
+    if args.gpu_spec != -2 and not torch.cuda.is_available():
```

Comment:
In your current argparse model, it's not possible for `gpu_spec` to take a value of `-2`. It's either `-1` (default) or `list[int]`. What would happen in the case that a user supplies `-g -2 -1 0 1`?